test/e2e/performanceprofile/functests/2_performance_update/memorymanager.go

```diff
@@ -0,0 +1,867 @@
+			mm2.memory = "200Mi"
+			mm2.hpgSize = profile.Spec.HugePages.Pages[1].Size
+			mm2.cpu = fmt.Sprintf("%d", len(available_node1_cpus)-2)
+			testPod2 := mm2.createPodTemplate(profile, true, targetNode)
+			// Initialize test pod, check if the pod uses Numa node 1
+			err = initializePod(testPod2)
+			Expect(err).ToNot(HaveOccurred(), "Unable to initialize pod")
+			numaZone, err = GetMemoryNodes(testPod2, targetNode)
+			// Expect numa node 1 to be used by pod
+			Expect(numaZone).To(Equal("1"))
+			Expect(err).ToNot(HaveOccurred(), "Pod's numa affinity is %s instead of %s", numaZone, "1")
+			//Delete pod
+			err = mm2.removePod(testPod2)
+			Expect(err).ToNot(HaveOccurred(), "failed to remove testpod")
+		})
+
+		AfterAll(func() {
+			By("Reverting the Profile")
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			currentSpec, _ := json.Marshal(profile.Spec)
+			spec, _ := json.Marshal(initialProfile.Spec)
+			// revert only if the profile changes.
+			if !bytes.Equal(currentSpec, spec) {
+				spec, err := json.Marshal(initialProfile.Spec)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testclient.Client.Patch(context.TODO(), profile,
+					client.RawPatch(
+						types.JSONPatchType,
+						[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+					),
+				)).ToNot(HaveOccurred())
+
+				By("Applying changes in performance profile and waiting until mcp will start updating")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+				By("Waiting when mcp finishes updates")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+			}
+		})
+	})
+})
+
+func checkPodEvent(testpod *corev1.Pod, podEventReason string) error {
+	podEvents, err := events.GetEventsForObject(testclient.Client, testpod.Namespace, testpod.Name, string(testpod.UID))
+	if err != nil {
+		testlog.Error(err)
+		return err
+	}
+	testlog.Infof("log pod %s/%s events due to failure", testpod.Namespace, testpod.Name)
+	reasons := []string{}
+	for _, event := range podEvents.Items {
+		testlog.Warningf("-> %s %s %s", event.Action, event.Reason, event.Message)
+		reasons = append(reasons, event.Reason)
+	}
+	truepodAffinity := false
+	for _, v := range reasons {
+		if v == podEventReason {
+			truepodAffinity = true
+		}
+	}
+	Expect(truepodAffinity).To(BeTrue())
+	return nil
+}
+
+func (mm MMPod) createPodTemplate(profile *performancev2.PerformanceProfile, gu bool, targetNode *corev1.Node) *corev1.Pod {
+	testNode := make(map[string]string)
+	testNode["kubernetes.io/hostname"] = targetNode.Name
+	mm.podV1Struct = pods.GetTestPod()
+	mm.podV1Struct.Namespace = testutils.NamespaceTesting
+	mm.namespace = testutils.NamespaceTesting
+	volumeName := fmt.Sprintf("hugepage-%si", mm.hpgSize)
+	mm.medium = fmt.Sprintf("HugePages-%si", mm.hpgSize)
+
+	if gu { // limits for guaranteed pods
```

Comment:
we can reduce using if here. I'd think of making the default as guaranteed and if gu false then fill for burstable.